File: 145-binary-tree-postorder-traversal/binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:

        result = []
        def postOrder(root):

            if not root:
                return root

            postOrder(root.left)
            postOrder(root.right)
            result.append(root.val)    

        postOrder(root)
        return result     

File: 145-binary-tree-postorder-traversal/binary-tree-postorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func postorderTraversal(root *TreeNode) []int {

    if root == nil {
        return []int{} 
    }

    left := postorderTraversal(root.Left)
    right := postorderTraversal(root.Right)

    return append(append(left, right...), root.Val)


    

}